src/accelerate/utils.py

```diff
@@ -738,6 +739,68 @@ def __post_init__(self):
         }
 
 
+@dataclass
+class FullyShardedDataParallelPlugin:
+    """
+    This plugin is used to enable fully sharded data parallelism.
+    """
+
+    from torch.distributed.fsdp.fully_sharded_data_parallel import BackwardPrefetch, CPUOffload, ShardingStrategy
+
+    sharding_strategy: Optional[ShardingStrategy] = field(
+        default=None,
+        metadata={"help": "Possible options are {}".format(ShardingStrategy.__members__)},
+    )
+    backward_prefetch: Optional[BackwardPrefetch] = field(
+        default=None,
+        metadata={"help": "Possible options are {}".format(BackwardPrefetch.__members__)},
+    )
+    auto_wrap_policy: Optional[Callable] = field(
+        default=None,
+        metadata={"help": "A callable specifying a policy to recursively wrap layers with FSDP"},
+    )
+    cpu_offload: Optional[CPUOffload] = field(
+        default=None,
+        metadata={"help": "Decides Whether to offload parameters and gradients to CPU."},
+    )
+    min_num_params: int = field(
+        default=None, metadata={"help": "FSDP's minimum number of parameters for Default Auto Wrapping."}
+    )
+    ignored_modules: Optional[Iterable[torch.nn.Module]] = field(
+        default=None,
+        metadata={"help": "A list of modules to ignore for FSDP."},
+    )
+
+    def __post_init__(self):
+        from torch.distributed.fsdp.fully_sharded_data_parallel import CPUOffload, ShardingStrategy
+        from torch.distributed.fsdp.wrap import default_auto_wrap_policy
+
+        if self.sharding_strategy is None:
+            self.sharding_strategy = ShardingStrategy(int(os.environ.get("FSDP_SHARDING_STRATEGY", 1)))
+
+        if self.cpu_offload is None:
+            if os.environ.get("FSDP_OFFLOAD_PARAMS", "false") == "true":
+                self.cpu_offload = CPUOffload(offload_params=True)
+            else:
+                self.cpu_offload = CPUOffload(offload_params=False)
+
+        if self.min_num_params is None:
+            self.min_num_params = int(os.environ.get("FSDP_MIN_NUM_PARAMS", 0))
+
+        if self.auto_wrap_policy is None:
+            if self.min_num_params > 0:
+                self.auto_wrap_policy = functools.partial(default_auto_wrap_policy, min_num_params=self.min_num_params)
+
+        self.fsdp_config = {
+            "sharding_strategy": str(self.sharding_strategy),
+            "cpu_offload": self.cpu_offload.offload_params,
+            "min_num_params": self.min_num_params,
+            "is_default_auto_wrap_policy": self.auto_wrap_policy.__name__ == "default_auto_wrap_policy",
+            "ignored_modules_present": self.ignored_modules is not None,
+            "backward_prefetch_present": self.backward_prefetch is not None,
+        }
```

Comment:
Why add this here, is this used anywhere? The dataclass in itself should be enough to contain the full config?

Reply:
It was used sort of "__repr__" to be printed when printing the accelerator state. removed this as the dataclass should be the full config.